Common/Python/PythonSlice.cs

```diff
@@ -0,0 +1,159 @@
+                )
+        {
+            _slice = slice;
+        }
+
+        /// <summary>
+        /// Gets the number of symbols held in this slice
+        /// </summary>
+        public new int Count
+        {
+            get { return _slice.Count; }
+        }
+
+        /// <summary>
+        /// Gets all the symbols in this slice
+        /// </summary>
+        public new IReadOnlyList<Symbol> Keys
+        {
+            get { return _slice.Keys; }
+        }
+
+        /// <summary>
+        /// Gets a list of all the data in this slice
+        /// </summary>
+        public new IReadOnlyList<BaseData> Values
+        {
+            get { return _slice.Values; }
+        }
+
+        /// <summary>
+        /// Gets the data corresponding to the specified symbol. If the requested data
+        /// is of <see cref="MarketDataType.Tick"/>, then a <see cref="List{Tick}"/> will
+        /// be returned, otherwise, it will be the subscribed type, for example, <see cref="TradeBar"/>
+        /// or event <see cref="Quandl"/> for custom data.
+        /// </summary>
+        /// <param name="symbol">The data's symbols</param>
+        /// <returns>The data for the specified symbol</returns>
+        public new dynamic this[Symbol symbol]
+        {
+            get
+            {
+                var data = _slice[symbol];
+
+                var dynamicData = data as DynamicData;
+                if (dynamicData != null)
+                {
+                    try
+                    {
+                        using (Py.GIL())
+                        {
+                            return _converter.InvokeMethod("Data", new[] { dynamicData.ToPython() });
+                        }
+                    }
+                    catch
+                    {
+                        // NOP
+                    }
+                }
+                return data;
+            }
+        }
+
+        /// <summary>
+        /// Determines whether this instance contains data for the specified symbol
+        /// </summary>
+        /// <param name="symbol">The symbol we seek data for</param>
+        /// <returns>True if this instance contains data for the symbol, false otherwise</returns>
+        public new bool ContainsKey(Symbol symbol)
+        {
+            return _slice.ContainsKey(symbol);
+        }
+
+        /// <summary>
+        /// Gets the data associated with the specified symbol
+        /// </summary>
+        /// <param name="symbol">The symbol we want data for</param>
+        /// <param name="data">The data for the specifed symbol, or null if no data was found</param>
+        /// <returns>True if data was found, false otherwise</returns>
+        public new bool TryGetValue(Symbol symbol, out dynamic data)
+        {
+            return _slice.TryGetValue(symbol, out data);
+        }
+
+        /// <summary>
+        /// Sets the converter
+        /// </summary>
+        public static void SetConverter()
```

Comment:
Is this class ever valid if this is not set? Perhaps instead of forcing consumers to poke this type, the type could handle this initialization itself? I've seen a similar pattern elsewhere, where we have a `SetXXX` method that accepts no arguments. Typically, with a `SetXXX` method, we'll pass the XXX to be set. It appears this is just a hook for some required static initialization.  Static constructors are often a good solution for static initialization issues.